src/rtos/task_notify_when_deleting.c

```diff
@@ -103,10 +144,16 @@ static void delete_hook_cb(ll_node_s_t* node, void* ignore) {
 }
 
 void task_notify_when_deleting_hook(task_t task) {
-  linked_list_s_t* ll = pvTaskGetThreadLocalStoragePointer(task, TLSP_IDX);
+  // if this task was subscribed to any other task deletion events, unsubscribe
+  linked_list_s_t* ll = pvTaskGetThreadLocalStoragePointer(task, SUBSCRIPTIONS_TLSP_IDX);
+  if (ll != NULL) {
+    linked_list_foreach(ll, unsubscribe_hook_cb, task);
+  }

```

Comment:
Why are you not freeing up the subscriptions linked list as `task` is dying?

Reply:
oh yeah. good point.